### DISCUSSION TITLE
Features request Group names and Group assignments

### DISCUSSION THREAD

**Discussion by pgScorpio** [jamulussoftware/jamulus#2242] [General]
**Intro:**
It is great we now can assign 8 different groups, but the more groups you use the harder it gets to remember which group number is which group.
I, and a lot of people I know, mostly work with large groups of 50-150 participants and in this case assigning groups (in most cases 6 or 7 different groups) is almost essential, but it is also a long and tedious job to set up the group assignments.
To make this job just a bit easier we often let all clients prefix their names with their group name, so when sorting by client name they are already sorted in the group assignment order. But still it's a tedious job to assign each client individually to a group with so many clients. (And actually every client would have to do this,)

**Feature request(s):**
To be able to assign custom names to the groups. (i.e. dropdown list/spinbox to select a group number + a textbox to set a custom name for that group in advanced settings?) *)

An option to clear all current group assignments.  (this option should also clear the assignments of all clients not currently online.)

An option to clear all current assignments to a specific group (also of clients not currently online.)

An option to **automatically** set the group assignment for new clients based on a client name wildcard (i.e. "Alt1-*") or instrument. (Using instrument when no name wildcard is defined or no name wildcard matches. Note that it should be possible to have multiple name wildcards assigned to each group.) This would also require the ability to set a default group for each instrument *)
Also an 'Auto reassign groups now' option should be available. (also for clients not currently online.)

An export/import option for Group names, Name wildcards and Instrument-Group assignments. *)

*) Manually editing a text file with the group names, Name wildcards and Instrument-Group assignments would be acceptable if we can export to a text file, edit the file with i.e. notepad and re-import it.

P.S. I think the audio guys of the Worldjam team would appreciate the **automatically** features too ;=))

**Comment by ann0see** [comment]
Idea: Select a group on a fader while holding shift. Then you’d click on all people you want to assign to the group.

**Reply by pgScorpio** [reply]
Great suggestion, but still a lot of work with many participants. Especially with large events, switching between different groups, there simply is no time to manually assign groups (think of events in Worldjam style, but with large choirs participating.).

Also with large choirs everybody needs to define it's own groups. With my proposal one person could create the auto assign definitions and distribute the file to import, so all choir members can import that file and they would all have the same group assignments automatically.

**Reply by pgScorpio** [reply]
P.S. Holding shift is also no option on touchscreen devices.

**Comment by ann0see** [comment]
Can you edit the configuration file (Jamulus.ini) to pre-set groups? Or the Jamulus mixing levels file?

**Reply by pgScorpio** [reply]
I don't know who you are asking and I don't know if I understand the question, but manually editing the Jamulus.ini is not something you would ask from the average user. That's why I request the export/import option.
Also, if it is about manually editing all the assignments, that's no option either. It's all about AUTOMATICALLY assigning groups based on name wildcards / instrument.

**Reply by ann0see** [reply]
> Manually editing a text file with the group names, Name wildcards and Instrument-Group assignments would be acceptable

That’s what I was referring to. There is a feature to export mixer settings. I‘m not sure which features it supports. Maybe it produces a text file, so there is the possibility that some of your requests could be already implemented.

**Reply by pgScorpio** [reply]
Ok. What I meant was, if editing in Jamulus is too difficult, one could export the group definitions to a text file. edit that text file. and import the modified text file.
